### PR TITLE
Add support php 8.x and composer 2.x

### PR DESCRIPTION
**Pull request by EddieOrbit**
### Added

- Support PHP `8.x`

### Changed

- Composer `2.x` is supported now

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I wrote unit tests for my code _(if tests is required for my changes)_
- [X] I have made changes in `CHANGELOG.md` file

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -16,16 +16,16 @@
         }
     ],
     "require": {
-        "php": "^7.3",
+        "php": "^7.3 || ^8.0",
         "ext-json": "*",
         "ext-mbstring": "*",
-        "avtocod/specs": "~3.70",
+        "avtocod/specs": "~3.77",
```
Comment (gomzyakov) [inline composer.json]: Why not `^3.0`? Actual version is `v3.101.0`

File: composer.json [hunk 2]
```diff
@@ -16,16 +16,16 @@
...
         "illuminate/support": "~6.0 || ~7.0 || ~8.0",
-        "ocramius/package-versions": "^1.4"
+        "composer/package-versions-deprecated": "^1.11"
     },
     "require-dev": {
-        "phpunit/phpunit": "~7.5 || ^8.0",
-        "phpstan/phpstan": "~0.12"
+        "phpunit/phpunit": "^8.5.4 || ^9.3",
+        "phpstan/phpstan": "~0.12.34"
```
Comment (gomzyakov) [inline composer.json:28]: Actual version is `1.4.7`

**UPD**: Comment not actual. This is a separate task. With `PHPStan v1+`, the diff will be very large.

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -48,27 +47,30 @@ jobs: # Docs: <https://git.io/JvxXE>
...
+        run: composer require --dev --update-with-all-dependencies --prefer-dist --ansi illuminate/support "${{ matrix.illuminate }}"
 
-      - name: Show most important packages versions
+      - name: Show most important package versions
         run: composer info | grep -e phpunit -e phpstan -e illuminate -e avtocod
 
       - name: Execute tests
         run: composer test-cover
 
+      - name: Execute tests with code coverage
```
Comment (gomzyakov) [inline .github/workflows/tests.yml:69]: This step is necessary?

On previous step we run: `composer test-cover`

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,16 @@ All notable changes to this project will be documented in this file.
...
 
+## UNRELEASED
+
+### Added
+
+- Support PHP `8.x`
+
+### Changed
+
+- Composer `2.x` is supported now
```
Comment (gomzyakov) [inline CHANGELOG.md:15]: What about `xdebug`?

**Comment by eldario** [comment]
Actual PR is #10

**Pull request closed without merging**